test/tls_test.go

```diff
@@ -248,6 +257,178 @@ func TestTLSClientCertificateSANsBasedAuth(t *testing.T) {
 	defer devc.Close()
 
 	// prod: Using SAN 'app.nats.prod' with all permissions.
+	//
+	//    Subject: CN = www.nats.io
+	//
+	//    X509v3 Subject Alternative Name:
+	//        DNS:app.nats.prod, DNS:*.app.nats.prod
```

Comment:
How do we resolve multiple entries?

Reply:
The cert defines the order in which multipe entries will be tried.  For example, in this case the cert has two SANs defined which will result in the following array always in this order:

```ruby
DNSNames = ["app.nats.prod", "*.app.nats.prod"]
```

We then try each until we find a SANs that is present in the config.  With the config below for example, `app.nats.prod` would be looked first but would fail since not present in the config and the SAN `*.app.nats.prod` would be used instead.  If neither of the SANs are present then in the config then the subject in the cert using the DN syntax would be used (`CN=www.nats.io` in this case).

```hcl
users [
  { user = "admin@nats.io" }    # Email is tried first, if no emails then other methods are tried
  { user = "*.app.nats.prod" }  # Use one of SANs if present
  { user = "CN=www.nats.io" }   # Defaul to use the subject in the cert if no other method available
]
```

Reply:
ok I think I follow.

Reply:
Yup, that's exactly the type of functionality I was looking for.  Thanks again!